### PR TITLE
[enhancement](decommission) speed up decommission process (#14028)

### LINKED ISSUE #13579: [Enhancement] decommission should safety drop be immediately when BE's tablets all in recycled state
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

When use decommission cmd to drop BE, some tablets on BE may be recycled, but BE's meta data holds these tablets. Because BE will be droped only when BE's meta data does not containes any tablet, so we always need to wait until trash is expired. So we can check BE's tablets, if all tablets of this BE is in recycled status, we can safety drop the BE immediately. 

### Solution

See Description above.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #14028: [Enhancement] decommission should safety drop be immediately when BE's tablets all in recycled state
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

When use decommission cmd to drop BE, some tablets on BE may be recycled, but BE's meta data holds these tablets. Because BE will be droped only when BE's meta data does not containes any tablet, so we always need to wait until trash is expired. So we can check BE's tablets, if all tablets of this BE is in recycled status, we can safety drop the BE immediately.

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by dutyu**
# Proposed changes

Issue Number: close #14028

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.58 seconds
 load time: 446 seconds
 storage size: 17179747628 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221107060320_clickbench_pr_40853.html

File: fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java [hunk 1]
```diff
@@ -84,10 +84,11 @@ private void runAlterJobV2() {
             }
 
             List<Long> backendTabletIds = invertedIndex.getTabletIdsByBackendId(beId);
-            if (backendTabletIds.isEmpty() && Config.drop_backend_after_decommission) {
+            if ((backendTabletIds.isEmpty() || Env.getCurrentRecycleBin().allTabletsInRecycledStatus(backendTabletIds))
```
[the remark this answers is not in the document]
Comment (dutyu) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java]: Appreciate for your suggestion, i have changed the code here.

[the remark this answers is not in the document]
Comment (dutyu) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java]: > `allTabletsInRecycledStatus` is expensive, maybe we should put `Config.drop_backend_after_decommission` as the first condition.

Would be appreciate if you spend some time to review it again.

**Review by cambyzju — COMMENTED** [review]
LGTM

**Review by hf200012 — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java [hunk 2]
```diff
@@ -70,6 +70,56 @@ public CatalogRecycleBin() {
         idToRecycleTime = Maps.newHashMap();
     }
 
+    public synchronized boolean allTabletsInRecycledStatus(List<Long> backendTabletIds) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java:73]: The `backendTabletIds` may contains millions of ids, which may cause a long run.

Comment (dutyu) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java:73]: Thanks, as we've just discussed, i will add a threshold to control this check and add some logs.

**Review by cambyzju — APPROVED** [review]
LGTM

**Pull request merged**